src/raftstore/store/peer_storage.rs

```diff
@@ -1470,6 +1471,166 @@ impl Storage for PeerStorage {
     }
 }
 
+/// Upgreade from v2.x to v3.x
+///
+/// For backward compatiblity, it needs to check whether there are any
+/// meta data in the raft cf of the kv engine, if there are, it moves them
+/// into raft engine.
+pub fn maybe_upgrade_from_2_to_3(
+    raft_engine: &DB,
+    kv_path: &str,
+    kv_db_opts: DBOptions,
+    kv_cfg: &config::DbConfig,
+) {
+    use crate::storage::CF_RAFT;
+    if !util::rocksdb_util::db_exist(kv_path) {
+        debug!("no need upgrade to v3.x");
+        return;
+    }
+
+    if DB::list_column_families(&kv_db_opts, kv_path)
```

Comment:
What happen if the number of cfs in `kv_db_opts` doesn't match the actual db cfs?

Reply:
`DBOptions` does not contain cf.